act/qc/bsrn_tests.py

```diff
@@ -750,3 +755,288 @@ def bsrn_comparison_tests(
                 test_assessment=test_assessment,
                 test_meaning=test_meaning,
             )
+
+        # Closure test
+        if test_options[6].lower() in test:
+            if (
+                direct_normal_SW_dn_name is None
+                or glb_diffuse_SW_dn_name is None
+                or gbl_SW_dn_name is None
+            ):
+                raise ValueError(
+                    'Must set keywords direct_normal_SW_dn_name, glb_diffuse_SW_dn_name, '
+                    f'gbl_SW_dn_name for {test_options[3]} test.'
+                )
+
+            if use_dask and isinstance(self._ds[gbl_SW_dn_name].data, da.Array):
+                sza = da.array(sza)
+                component_sum = self._ds[gbl_SW_dn_name].values + (
+                    self._ds[direct_normal_SW_dn_name].values * np.cos(np.radians(sza))
+                )
+
+                index_lsz = da.where((component_sum > 50.0) & (sza <= 75.0), True, False)
+                index_hsz = da.where(
+                    (component_sum > 50.0) & (sza > 75.0) & (sza < 93.0), True, False
+                )
+
+                value = (self._ds[glb_diffuse_SW_dn_name].values / component_sum) - 1.0
+
+                index_1 = da.where((value >= 0.08) & index_lsz, True, False)
+                index_2 = da.where((value >= 0.15) & index_hsz, True, False)
+
+                index = (index_1 | index_2).compute()
+
+            else:
+                component_sum = self._ds[gbl_SW_dn_name].values + (
+                    self._ds[direct_normal_SW_dn_name].values * np.cos(np.radians(sza))
+                )
+
+                index_lsz = (component_sum > 50.0) & (sza <= 75.0)
+                index_hsz = (component_sum > 50.0) & (sza > 75.0) & (sza < 93.0)
+
+                value = (self._ds[glb_diffuse_SW_dn_name].values / component_sum) - 1.0
+
+                index_1 = (value >= 0.08) & index_lsz
+                index_2 = (value >= 0.15) & index_hsz
+
+                index = index_1 | index_2
+
+            test_meaning = "Closure test indicating value outside of expected range"
+            self._ds.qcfilter.add_test(
+                direct_normal_SW_dn_name,
+                index=index,
+                test_assessment=test_assessment,
+                test_meaning=test_meaning,
+            )
+            self._ds.qcfilter.add_test(
+                glb_diffuse_SW_dn_name,
+                index=index,
+                test_assessment=test_assessment,
+                test_meaning=test_meaning,
+            )
+            self._ds.qcfilter.add_test(
+                gbl_SW_dn_name,
+                index=index,
+                test_assessment=test_assessment,
+                test_meaning=test_meaning,
+            )
+
+    def k_test(
```

Comment:
@kenkehoe since there's ktest in scipy and this one seems specific to bsrn QC, it might be beneficial to have a more descriptive name so users don't think this is a generic k-test like in scipy.

Reply:
OK good point. I will rename.